src/amo/components/RatingsByStar/index.js

```diff
@@ -86,85 +91,100 @@ export class RatingsByStarBase extends React.Component<InternalProps> {
     );
   }
 
+  handleRouting(star: string) {
+    const { addon, location, history, clientApp, siteLang } = this.props;
+
+    invariant(addon, 'addon was unexpectedly empty');
+
+    history.push(
+      `/${siteLang}/${clientApp}${reviewListURL({
+        addonSlug: addon.slug,
+        score: star,
+        src: location.query.src,
+      })}`,
+    );
+  }
+
   render() {
-    const { addon, errorHandler, i18n, groupedRatings, location } = this.props;
+    const { addon, errorHandler, i18n, groupedRatings } = this.props;
     const loading = (!addon || !groupedRatings) && !errorHandler.hasError();
+    const self = this;
+
+    const getLinkTitle = (star, count = '') => {
+      switch (star) {
+        case '5':
+          return i18n.gettext(`Read all ${count} five-star reviews`);
+        case '4':
+          return i18n.gettext(`Read all ${count} four-star reviews`);
+        case '3':
+          return i18n.gettext(`Read all ${count} three-star reviews`);
+        case '2':
+          return i18n.gettext(`Read all ${count} two-star reviews`);
+        case '1':
+          return i18n.gettext(`Read all ${count} one-star reviews`);
+        default:
+          return '';
+      }
+    };
 
-    const linkTitles = {
-      /* eslint-disable quote-props */
-      '5': i18n.gettext('Read all five-star reviews'),
-      '4': i18n.gettext('Read all four-star reviews'),
-      '3': i18n.gettext('Read all three-star reviews'),
-      '2': i18n.gettext('Read all two-star reviews'),
-      '1': i18n.gettext('Read all one-star reviews'),
-      /* eslint-enable quote-props */
+    const createTableRow = (star) => {
+      let starCount;
+      let starCountNode;
+
+      if (!errorHandler.hasError()) {
+        if (groupedRatings) {
+          starCount = groupedRatings[star];
+        }
+
+        starCountNode = i18n.formatNumber(starCount || 0);
+      }
+
+      const rowProps =
+        loading || !addon
+          ? {}
```

Comment:
I'm not sure we have a test for this logic. I.e., a test that asserts that `onClick` and `title` do not exist on the row when either `loading` is `true` or when we have no `addon`.

Reply:
I think we can check the `onClick` prop for row in test `renders a loading state without an add-on`. Is this correct?

Reply:
Yes, we could add those assertions to that test, but it looks like you haven't done that in the last commit.

Reply:
Sorry, just forgot to add this one. I will update this in the next commit.